File: Creational - Abstract Factory/Python/concrete_factory/concrete_factory_2.py
from abstract_factory.abstract_factory import AbstractFactory
from abstract_product.abstract_product_a import AbstractProductA
from abstract_product.abstract_product_b import AbstractProductB
from concrete_product.concrete_product_a2 import ConcreteProductA2
from concrete_product.concrete_product_b2 import ConcreteProductB2


class ConcreteFactory2(AbstractFactory):
    def create_product_a(self) -> AbstractProductA:
        return ConcreteProductA2()

    def create_product_b(self) -> AbstractProductB:
        return ConcreteProductB2()

File: Creational - Abstract Factory/Typescript/concrete-factory/concrete-factory-2.ts
import { AbstractFactory } from "../abstract-factory/abstract-factory.ts";
import { AbstractProductA } from "../abstract-product/abstract-product-a.ts";
import { AbstractProductB } from "../abstract-product/abstract-product-b.ts";
import { ConcreteProductA2 } from "../concrete-product/concrete-product-a2.ts";
import { ConcreteProductB2 } from "../concrete-product/concrete-product-b2.ts";


export class ConcreteFactory2 implements AbstractFactory {
  public createProductA(): AbstractProductA {
      return new ConcreteProductA2();
  }

  public createProductB(): AbstractProductB {
      return new ConcreteProductB2();
  }
}